### ISSUE TITLE
Data Type 0x25 uint48 seems to be wrong

### ISSUE THREAD

**Issue by pipiche38** [fairecasoimeme/ZiGate#229]
When receiving an attribute of Data Type 0x25, the corresponding length is 0x0008, while it should be 0x0006 , as 0x25 is Unsigned 48 Bits 6 x 8 bits.

In addition the Data attributes (which has a length of 8) seems to be truncated and only the 4 first bytes are correct !

The issue is reported while receiving Cluster  0x0101 Attribute 0x0508 of the Xiaomi Vibration.

The end result is that we are unable to compute the Vibration orientation !

**Comment by schrodingersket** [comment]
Related to #223 perhaps? (:

**Comment by pipiche38** [comment]
You might be right, as I did further investigation and indeed even for Data Type 0x2a we got a size of 0x0004 instead of 0x0003 as we are expecting a U24bits !

It was however fixed in 3.0f !

@fairecasoimeme

**Comment by fairecasoimeme** [comment]
It's normal, I cast Uint48 with Uint32.
But you're right, there are inconsistency between size and datas. I'll fix that.
thanks

**Comment by pipiche38** [comment]
Is there any impact on this fix to the 0x2A Data Type ?

**Comment by fairecasoimeme** [comment]
No :( no change for 0x2A. How could I test ? do you know a device which use this data type ?

**Comment by pipiche38** [comment]
Basically Cluster 0x0702 / Attribut 0x0400
As of today on Firmware 3.1a

Here is what I'm getting:

Attribute Type: 0x2a
Attribute Size : 0x0004
Attribute Value: 0000011d

And that is where I'm concern, as If I take the logic you put now, I would expect

Attribute Size: 0x0003
Attribute Value: 00011d

Which is quiet different

**Comment by pipiche38** [comment]
I'll try to test it and revert .
Main issue is the Salus Plug I have is on my prod network ... So I have some logistic to be done

**Comment by ISO-B** [comment]
@pipiche38 Did you managed to work solution to this?

**Comment by pipiche38** [comment]
The 0x25 uint48 is indeed fixed by 3.1b I got to test. So it is good.

**Comment by aargollo** [comment]
Hi Guys!
I would like to check if somebody have tested the last Zigate Version (v3.1.c) Data type 25 with Attribute CurrentSummationDelivered (0x0000)?
I observed that the zigate show 8 bytes length , over Data Type 25, and is discarding the measurement important information. Alway represents all those bytes as 0x00. As the PIC attached bellow.

![image](https://user-images.githubusercontent.com/4733441/92667018-2fb25080-f2e1-11ea-874f-7f1ce9a001c4.png)


I got messages over the air, via wireshark and I observed it:
**ZCL Command :** 0x0a - Report Command Attribute 
Attribute filed:
**Attribute:** Current Summation Delivered (0x0000)
**Data Type:** 48-bits Unsigned Integer
**Data:** Measurement value (6 bytes).
I believed that Data Type 25 should be have this format: .... 25, **0x2, 0x10, 0x2, 0x16,** + **6 bytes** + RSSI Byte and EOT (0x03).

According Zigate Protocol (v.3) Report Attribute Response should be have:
<Attribute data   type: uint8_t>
<Size Of the attributes in bytes:   uint16_t>
<Data byte list : stream of   uint8_t>

![image](https://user-images.githubusercontent.com/4733441/92667224-b5ce9700-f2e1-11ea-9213-41e1574b55ae.png)

**Comment by pipiche38** [comment]
Do you have this problem only with ZGUI ? 
As I do not see any issue with the current 31c which report perfectly on Data Type 0x25

**Comment by aargollo** [comment]
Hi my friend. It´s a good question, I didn´t do it before, but today I did double check in the ZGUI and I observed the same behavior as I posted yesterday. As you can see bellow:

 according Wireshark the data value from Data Typ2 25 - Attribute CurrentSummationDelivered (0x0000), the data value (6bytes), from message (SQN e9 - 233) is: **2c02**
![image](https://user-images.githubusercontent.com/4733441/92769660-954d1e00-f36f-11ea-9e79-d7d82e4db05a.png)

But in the ZGUI the same message (SQN e9 - 233) all those bytes in the Data type 25 are all those bytes as zero (0x00).
![image](https://user-images.githubusercontent.com/4733441/92770330-3a67f680-f370-11ea-93e1-7bd0cfd230c8.png)

**Comment by pipiche38** [comment]
It might be the case that ZGUI is not good to handle the Data Type 0x25 which was fixed in 31b/31c firmware. Have you tried with a python library or something else ?

**Comment by aargollo** [comment]
HI Fred,
I have include the 3.1.c version.
I got the packages from my Node-red interface and I saw the same results,
all those 8 bytes with 0x00 value.

Best Regards,
Alex


Em qui, 10 de set de 2020 14:34, Pipiche <notifications@github.com>
escreveu:

**Comment by aargollo** [comment]
HI guys, Sorry. I don´t know why, but after new tests I observed that my Zigate had inside a old firmware, since I did the 3.1.c update.  and I could check that everything is ok. I have tests Current Summation Delivered (Data type 25- 48 bits) and Instantaneous demand (data type 42 - 24 bits).
**Data flow:**
![image](https://user-images.githubusercontent.com/4733441/92960229-a8c9b900-f443-11ea-8c4c-c3f22247c2cf.png)

**OTA log via wireshark**

 
--
